test/integration/controller/core/workload_controller_test.go

```diff
@@ -62,12 +62,15 @@ var _ = ginkgo.Describe("Workload controller", func() {
 
 	ginkgo.AfterEach(func() {
 		clusterQueue = nil
+		queue = nil
 		updatedQueueWorkload = kueue.Workload{}
 		updatedCQ = kueue.ClusterQueue{}
-		gomega.Expect(framework.DeleteNamespace(ctx, k8sClient, ns)).To(gomega.Succeed())
```

Comment:
Just move this to each tests for after this PR, the clusterQueue can only be deleted after the corresponding running workloads finished. Now I'll delete the namespace first then delete the clusterQueues, then these clusterQueues can be deleted successfully. 

If we call `DeleteNamespace` here, according to the ginkgo rules 
```
When multiple AfterEach nodes are defined in nested Container nodes the innermost 
AfterEach node closures are run first.
```
ClusterQueue will be deleted firstly before workloads, it's not right.

Reply:
We can see errors In scheduler_test.go for clusterQueue can't be cleaned up successfully, so when we try to create a same name clusterqueue, it failes. I didn't deep dive for the reason but did some refactoring works to organize the tests better.

Reply:
I think we should look into those errors, can you post them? the order shouldn't matter if the controller is properly handling the finalizers.

Reply:
Placing "sleep" isn't a good practice and will cause flakes.

My suggestion is to call `gomega.Eventually` to wait for the cluster scoped resources (like ClusterQueues and ResourceFlavors) to be deleted (by doing a Get and wait for NotFound error).